src/katgpucbf/xbgpu/engine.py

```diff
@@ -346,7 +356,7 @@ def __init__(
             self.max_active_chunks, name="recv_data_ringbuffer", task_name=RECV_TASK_NAME, monitor=monitor
         )
         free_ringbuffer = spead2.recv.ChunkRingbuffer(n_free_chunks)
-        layout = recv.Layout(
+        self._src_layout = recv.Layout(
```

Comment:
I'd done this because it seemed better to keep some sort of parity with how fgpu does its `recv_chunks` async generator (`chunk_sets`).
* I didn't quite follow why the `_receiver_loop` necessarily needed it as an arg, as it's an attribute anyway?
* So I met in the middle, made it an attribute but kept the function signatures of the processing loops the same (empty).